src/renderers/webgl/WebGLRenderStates.js

```diff
@@ -101,7 +101,13 @@ function WebGLRenderStates() {
 
 	function dispose() {
 
-		renderStates = new WeakMap();
+		renderStates.forEach( function ( _, scene ) {
```

Comment:
I think this can be a simple loop:

```js
for ( let scene in renderStates ) {

    scene.removeEventListener( 'dispose', onSceneDispose );

}
```

Reply:
:+1: